### PR TITLE
Zlibwrapper

### PR DESCRIPTION
**Pull request by inikep**

### CODE REVIEW

File: zlibWrapper/zstd_zlibwrapper.h [hunk 1]
```diff
@@ -39,8 +39,9 @@ int ZWRAP_isUsingZSTDcompression(void);
 
 /* Changes a pledged source size for a given compression stream.
    It will change ZSTD compression parameters what may improve compression speed and/or ratio.
-   The function should be called just after deflateInit(). It's only helpful when data is compressed in blocks. 
-   There will be no change in case of deflateInit() immediately followed by deflate(strm, Z_FINISH) 
+   The function should be called just after deflateInit() or deflateReset() and before deflate() or deflateSetDictionary().
+   It's only helpful when data is compressed in blocks. 
+   There will be no change in case of deflateInit() or deflateReset() immediately followed by deflate(strm, Z_FINISH) 
    as this case is automatically detected.  */
 int ZWRAP_setPledgedSrcSize(z_streamp strm, unsigned long long pledgedSrcSize);
```
Comment (Cyan4973) [inline zlibWrapper/zstd_zlibwrapper.h]: Is `ZWRAP_setPledgedSrcSize()` still useful as a public interface ?
I had the feeling you solved the situation within detection of `Z_FINISH`, making this additional interface less relevant.

File: zlibWrapper/zstd_zlibwrapper.c [hunk 2]
```diff
@@ -203,15 +203,26 @@ ZEXTERN int ZEXPORT z_deflateInit2_ OF((z_streamp strm, int level, int method,
 }
 
 
+int ZWRAP_deflateResetWithoutDict(z_streamp strm)
```
Comment (Cyan4973) [inline zlibWrapper/zstd_zlibwrapper.c]: See comment on Asana regarding choice of function name

File: zlibWrapper/zstd_zlibwrapper.h (hunk 1 above)
Comment (inikep) [inline zlibWrapper/zstd_zlibwrapper.h]: It's optional but it will help for e.g. `write_large_blob_data()` which is compressed in 16KB blocks.

**Comment by Cyan4973** [comment]
Valgrind detected something : 
https://travis-ci.org/facebook/zstd/jobs/163128405

**Comment by inikep** [comment]
It should be fixed now.

**Pull request merged**